src/go/rpk/pkg/cli/cmd/common/common_test.go

```diff
@@ -351,106 +414,261 @@ func TestKafkaAuthConfig(t *testing.T) {
+5FWiLjwxu1UzElnCTpay+y4snQpRDf46JMZ3UkW66GNCZulImIU9kalWOuPE3bh4
+bsyMIZfcs4exHOV0g2EenLk=
+-----END CERTIFICATE-----`
+
+	keyContents := `-----BEGIN RSA PRIVATE KEY-----
+MIIEpQIBAAKCAQEA1Ti8hqVJg5vQ66LZloOMPFbTvkvZ6tK+lmDtBbGnlZOYNaCW
+zL+xfGPRq5vUcO7p3Jk8VIg5zhO6eMz7RpTtotHtoruT2wLGqzwV1HHhXABD8RG4
+nPOs63Dmpv9avfjkrtWZCo7ed8iLFziaKmCyxZYAVQ4K/PeECh2oIMrCuxBLeKec
+botRQQODJ5KmOilK57uW4cJocvQrA0/8CXMs2HFMk/wyooOAVfzDyhfCUhybM5Q7
+ZE+rTw52yl7mD2x+GI6jS7NG8+LHueuDh7qn7NICdSnoFMbFIvh4dFANNu1cawC/
+VmidsOkBDCFKCzPKnIbZh4aRqZ0fxxLqE7LrJwIDAQABAoIBAQDQ3yuPmwtQ6arX
+qkgMsgEGeugiWpu29YvONFT8ZvQMCvHoVtBi8sYjXIVg3t5VYzWk7Fe1V12JCrp4
+7BSbJ/lCrvNjnu1Qdn+37rxTyNtDDN+BoCKBXhPe8FKC9VMnFlKvEn9BYIN+Q+49
+aS1cpi16cV8R8xfAh5fJcRPqS7ZHF/1rhUqkcoV956h5RtQy3k/BIVpJydbrb17U
+xEhiCFyRbCIEEnX2hTRFrRt4hEIwa2YceWMsY8ddCngtS/i4UGLsaYLCMrTgtvwS
+H9uI1q0gL5tcp7qo4MQo9BRtbiazV70MofabF19tapDlbPoqcggN/sTi4VQzFWsx
+YKShZhIBAoGBAPA5ZaH6pg76O+E47kPbugx2V6y8YongvBOe+MXkwXG3samLUhN9
+63th3qhDOqqwUN8FgJYqATbn6lsvVnIhoJ1EK126grVeZme7FCTswjHKd2gaVAaz
+GBJ7ejMMoHzw1IGSXxnZ69XFN4xUSAiCeq58LWl3PJmtpt+vZ2ZZrA+HAoGBAOM5
+YJP7YzGPG/OX4EVttGzcMCigTUTKZwH4aNTzJm2KDHiS3uiW6XMgMsY8H057XZbT
+QjB7QnokRUuVqD/Zugh2bHZcHiJNT9re/TZwqXPXh2B4fY0ZOSAImMpsXWdfJSKK
+anIroNF4v5gw+4zqtEBE8Zc7Ct+gbvfPp9qdbu9hAoGAXkAWyQufdYbmUYJVsVgX
+UeZolcQ/4RrEj+oybupGn4hT81JPPIiOCJWol1nxPaD5ydbN0ZzfZxxszaPwBc19
+x9ZEMX0I5YIJKa+zwp0FwCVQ3g5eY1aHHlFF65uLqBmRNtkn6OugZPoAxlUXAge3
+fJgJ9TQsGZuROngGWJjcMicCgYEAxPd12pFt2QX+6tfalxST9FGihXT/xgPV6wVU
+ilQEGawzR0m5ZNF8qEle+iwfzz5tUFLs623NoGdUkkK2yDKKas+NEcSkcoOmF0p5
+IPnkSgCo311TKD6XIEeTetUY2oTFgf2ObE2ZaDtNijXbuLmzaorZCYkq0dMWnkYp
+cP5LrcECgYEA4+Sk/uND5dDc1TXRKGaiu9t2DscUpQCwk5PVOMMA0iwBJBodt85o
+nDpxxecx1Mh1+0V46bnrYrIrAGurqCbQN8B5EsB9dcLZ3bg86KRwP3ZbQNxKPIqo
+T16cNmHSk5jIiR6odFmV6KPjvXhjFTUYxOIjFIWNItOhXBrBxG3NyVE=
+-----END RSA PRIVATE KEY-----`
+
+	certVarName := "RP_TEST_CERT"
+	keyVarName := "RP_TEST_KEY"
+	truststoreVarName := "RP_TEST_TRUSTSTORE"
+
 	tests := []struct {
 		name           string
 		keyFile        string
 		certFile       string
 		truststoreFile string
-		before         func()
+		before         func(*testing.T, afero.Fs)
 		cleanup        func()
-		expected       *config.TLS
+		defaultVal     func() (*config.TLS, error)
 		expectedErrMsg string
 	}{{
-		name:     "it should return a nil config if none are set",
-		expected: nil,
-	}, {
-		name:           "it should fail if truststoreFile is empty",
-		certFile:       "cert.pem",
-		keyFile:        "key.pem",
-		expectedErrMsg: "--tls-truststore is required to enable TLS",
+		name: "it should return the default value provided if none are set",
+		defaultVal: func() (*config.TLS, error) {
+			return &config.TLS{
+				CertFile:       "default-cert.pem",
+				KeyFile:        "default-key.pem",
+				TruststoreFile: "default-trust.pem",
+			}, nil
+		},
+		before: func(t *testing.T, fs afero.Fs) {
+			require.NoError(t, afero.WriteFile(fs, "default-cert.pem", []byte(certContents), 0755))
```

Comment:
I don't think `require.NoError` is meaningful here since this is writing to an in-memory filesystem we are defining, and if this does fail, the tests below will fail.

Reply:
I wonder if this would be good to add to the new `testfs` package, as
```go
func ExtendFromMap(fs afero.Fs, m map[string]Fmode)
```

Reply:
What would it do?

Reply:
It'd basically be a different way to define the same thing you're doing here,
```go
testfs.ExtendFromMap(fs, map[string]Fmode{
  "default-cert.pem": {0755, certContents},
   ...
})
```